### DISCUSSION TITLE
How to use a generic Control? FieldArray type?

### DISCUSSION THREAD

**Discussion by mrdjohnson** [react-hook-form/react-hook-form#12402] [Q&A]
I have two objects, both of which extend: `{ params: Param[] }`

I want to have a component that handles the field array for params but there does not seem to be a `FieldArray<Param[]>` type I can pass down. I an tell the component to accept  `{ control: Control<{params: Param[]}> }`, but I get an error when I try to pass in a control because the full field list is different. 

Any help would be appreciated! 

(I'm trying to do the router with hook forms approach but passing down the Control instead of going through a state manager's context. )

**Comment by BrendanC23** [comment]
> I get an error when I try to pass in a control because the full field list is different

There's not a great way to do this. See [this issue](https://github.com/react-hook-form/react-hook-form/issues/11617) for more details. Using `Control` is the best way, but the fields need to match exactly. [This](https://github.com/orgs/react-hook-form/discussions/7354#discussioncomment-2786306) is the best workaround I've found. [This PR](https://github.com/react-hook-form/react-hook-form/pull/12284) might also be useful, though I haven't tested it myself.
